drivers/ieee802154/ieee802154_cc13xx_cc26xx_subg.c

```diff
@@ -0,0 +1,705 @@
+			continue;
+		}
+
+		/* TODO: handle RX acknowledgment */
+		r = 0;
+		goto out;
+
+	} while (retry-- > 0);
+
+	LOG_DBG("Failed to TX");
+	r = -EIO;
+
+out:
+	(void)ieee802154_cc13xx_cc26xx_subg_rx(dev);
+	k_mutex_unlock(&drv_data->tx_mutex);
+	return r;
+}
+
+static inline uint8_t ieee802154_cc13xx_cc26xx_subg_convert_rssi(int8_t rssi)
+{
+	if (rssi > CC13XX_CC26XX_RECEIVER_SENSITIVITY +
+	    CC13XX_CC26XX_RSSI_DYNAMIC_RANGE) {
+		rssi = CC13XX_CC26XX_RECEIVER_SENSITIVITY +
+		       CC13XX_CC26XX_RSSI_DYNAMIC_RANGE;
+	} else if (rssi < CC13XX_CC26XX_RECEIVER_SENSITIVITY) {
+		rssi = CC13XX_CC26XX_RECEIVER_SENSITIVITY;
+	}
+
+	return (255 * (rssi - CC13XX_CC26XX_RECEIVER_SENSITIVITY)) /
+	       CC13XX_CC26XX_RSSI_DYNAMIC_RANGE;
+}
+
+static void ieee802154_cc13xx_cc26xx_subg_rx_done(
+	struct ieee802154_cc13xx_cc26xx_subg_data *drv_data)
+{
+	struct net_pkt *pkt;
+	uint8_t len;
+	int8_t rssi;
+	uint8_t *sdu;
+
+	for (int i = 0; i < CC13XX_CC26XX_NUM_RX_BUF; i++) {
+		if (drv_data->rx_entry[i].status == DATA_ENTRY_FINISHED) {
+			len = drv_data->rx_data[i][1];
+			sdu = &drv_data->rx_data[i][3];
+			rssi = sdu[len - 2];
+			len -= 2;
+
+			LOG_DBG("Received: len = %u, rssi = %d", len, rssi);
+
+			pkt = net_pkt_rx_alloc_with_buffer(
+				drv_data->iface, len, AF_UNSPEC, 0, K_NO_WAIT);
+			if (!pkt) {
+				LOG_WRN("Cannot allocate packet");
+				continue;
+			}
+
+			if (net_pkt_write(pkt, sdu, len)) {
+				LOG_WRN("Cannot write packet");
+				net_pkt_unref(pkt);
+				continue;
+			}
+
+			drv_data->rx_entry[i].status = DATA_ENTRY_PENDING;
+
+			/* TODO determine LQI in PROP mode */
+			net_pkt_set_ieee802154_lqi(pkt, 0xff);
+			net_pkt_set_ieee802154_rssi(
+				pkt,
+				ieee802154_cc13xx_cc26xx_subg_convert_rssi(rssi));
+
+			if (net_recv_data(drv_data->iface, pkt)) {
+				LOG_WRN("Packet dropped");
+				net_pkt_unref(pkt);
+			}
+
+		} else if (drv_data->rx_entry[i].status ==
+			   DATA_ENTRY_UNFINISHED) {
+			LOG_WRN("Frame not finished");
+			drv_data->rx_entry[i].status = DATA_ENTRY_PENDING;
+		}
+	}
+}
+
+static int ieee802154_cc13xx_cc26xx_subg_start(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+	return 0;
+}
+
+static int ieee802154_cc13xx_cc26xx_subg_stop(const struct device *dev)
+{
+	struct ieee802154_cc13xx_cc26xx_subg_data *drv_data = get_dev_data(dev);
+	RF_Stat status;
+
+	status = RF_flushCmd(drv_data->rf_handle, RF_CMDHANDLE_FLUSH_ALL, 0);
+	if (!(status == RF_StatCmdDoneSuccess || status == RF_StatSuccess
+	      || status == RF_StatRadioInactiveError || status == RF_StatInvalidParamsError)) {
+		LOG_DBG("Failed to abort radio operations (%d)", status);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int
+ieee802154_cc13xx_cc26xx_subg_configure(const struct device *dev,
+					enum ieee802154_config_type type,
+					const struct ieee802154_config *config)
+{
+	return -ENOTSUP;
+}
+
+uint16_t ieee802154_cc13xx_cc26xx_subg_get_subg_channel_count(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return 11;
```

Comment:
add a comment about this number

Reply:
Jeez - yeah, that one totally got by me. Good catch!